Source/DafnyRuntime/DafnyRuntimeRust/src/tests/mod.rs

```diff
@@ -0,0 +1,411 @@
+        }
+    }
+
+    #[test]
+    fn test_const_this_in_datatype() {
+        let w: Rc<WithConstInitializer> = WithConstInitializer::_new(2);
+
+        assert_eq!(w.x, 2);
+        assert_eq!(w._z(), 1);
+        assert_eq!(w._itself().w.x, 2);
+    }
+
+    #[test]
+    fn test_sequence() {
+        let values = Rc::new(vec![1, 2, 3]);
+        let seq = Sequence::<i32>::new_array_sequence(&values);
+        assert_eq!(seq.node_count(), 1);
+        assert_eq!(seq.cardinality(), 3);
+        assert_eq!(seq.to_array(), values);
+    
+        // Create a concat array, wrap it into a lazy one, get the i-th element,
+        // and verify that this operation flattened the array
+        let left = Sequence::<i32>::new_array_sequence(&Rc::new(vec![1, 2, 3]));
+        let right = Sequence::<i32>::new_array_sequence(&Rc::new(vec![4, 5, 6]));
+        let concat = Sequence::<i32>::new_concat_sequence(&left, &right);
+
+        let lazy = Sequence::<i32>::new_lazy_sequence(&concat);
+        assert_eq!(lazy.node_count(), 4);
+        assert_eq!(lazy.cardinality(), 6);
+        match lazy.borrow() {
+            Sequence::LazySequence { boxed, .. } =>
+            match (&*boxed.borrow()).borrow() {
+                Sequence::ConcatSequence { is_string, node_count, length, .. } =>
+                {
+                    assert_eq!(*is_string, false);
+                    assert_eq!(*node_count, 3);
+                    assert_eq!(*length, 6);
+                },
+                _ => panic!("This should never happen")
+            },
+            _ => panic!("This should never happen")        
+        }
+        let value = lazy.select(0);
+        assert_eq!(value, 1);
+        match lazy.borrow() {
+            Sequence::LazySequence { boxed, .. } =>
+            match (&*boxed.borrow()).borrow() {
+                Sequence::ArraySequence { values, .. } =>
+                  assert_eq!(values, &Rc::new(vec![1, 2, 3, 4, 5, 6])),
+                _ => panic!("This should never happen")
+            },
+            _ => panic!("This should never happen")        
+        }
+    }
+
+    #[test]
+    fn test_native_array_pointer() {
+        let values: *const Vec<i32> = Box::into_raw(Box::new(vec![1, 2, 3]));
+        // allocate another vec of size 100
+        let values2: *const Vec<i32> = allocate(Box::new(vec![0; 100]));
+
+        // Verify that the length of values is 3
+        assert_eq!(unsafe{(*values).len()}, 3);
+        // If we change the first element to 4, we should read it as being 4 again
+        unsafe{*(*(values as *mut Vec<i32>)).get_unchecked_mut(0) = 4};
+        assert_eq!(unsafe{*(*values).get_unchecked(0)}, 4);
+
+        deallocate(values);
+    }
+
+    fn Test2(x: bool, y: &mut bool, z: &mut bool) {
+        *y = !x;
+        *z = !x || *y;
+    }
+    fn Test3() { 
+      let mut y = true; let mut z = true;
+      Test2(true, &mut y, &mut z);
+      let mut i; let mut j;
+      i = y;
+      j = z;
+    }
```

Comment:
These look like random code written to experiment with semantics, and AFAICT they aren't actually part of any tests, so we should remove them.

Reply:
I removed the entire file from the index and kept it locally until we use it for proper unit testing.